File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> pq = new PriorityQueue<>((a, b) -> a - b);
        for(int num: scoville) {
            pq.offer(num);
        }
        
        int answer = 0;
        while(!pq.isEmpty()){
            int num1 = pq.poll();   // 가장 맵지 않은 음식 스코빌 지수
            
            while(num1 < K && pq.size() > 0) {
                int num2 = pq.poll();   // 두 번째로 맵지 않은 음식의 스코빌 지수
                
                pq.offer(num1 + (num2 * 2));
                num1 = pq.poll();
                answer++;
            }
            
            if(num1 < K) {
                return -1;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    
    heapq.heapify(scoville)
    
    while scoville[0] < K:  # 가장 작은 값이 K보다 작으면 계속
        if len(scoville) < 2:  # 섞을 음식이 없으면 실패
            return -1
        
        s1 = heapq.heappop(scoville)
        s2 = heapq.heappop(scoville)
        
        newS = s1 + (s2 * 2)
        heapq.heappush(scoville, newS)
        answer += 1
    
    return answer